File: VB/ChartsDemo.Wpf/ViewModels/HumidityData.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Xml.Serialization

Namespace ChartsDemo
    <XmlRoot("HumidityData")>
    Public Class HumidityData
        Inherits List(Of HumidityInfo)


        Private Shared londonData_Renamed As List(Of HumidityInfo) = Nothing
        Public Shared ReadOnly Property LondonData() As List(Of HumidityInfo)
            Get
                If londonData_Renamed Is Nothing Then
                    Dim s = New XmlSerializer(GetType(HumidityData))
                    londonData_Renamed = DirectCast(s.Deserialize(DataLoader.LoadFromResources("/Data/LondonHumidity.xml")), List(Of HumidityInfo))
                End If
                Return londonData_Renamed
            End Get
        End Property
    End Class
    Public Class HumidityInfo
        Public Property Timestamp() As Date
        Public Property Value() As Integer
    End Class
End Namespace

File: CS/ChartsDemo.Wpf/ViewModels/HumidityData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Xml.Serialization;

namespace ChartsDemo {
    [XmlRoot("HumidityData")]
    public class HumidityData : List<HumidityInfo> {
        static List<HumidityInfo> londonData = null;
        public static List<HumidityInfo> LondonData {
            get {
                if(londonData == null) {
                    var s = new XmlSerializer(typeof(HumidityData));
                    londonData = (List<HumidityInfo>)s.Deserialize(DataLoader.LoadFromResources("/Data/LondonHumidity.xml"));
                }
                return londonData;
            }
        }
    }
    public class HumidityInfo {
        public DateTime Timestamp { get; set; }
        public int Value { get; set; }
    }
}
